prepare 2.0.0 release

owaddell-beckman opened the pull request:
Still a fair bit to do here:
- [X] sketch GitHub issue template (have folks include output of `swish-build --info`)
- [X] sketch GitHub pull-request template
- [X] create dummy private repo to test drive issue / pull-request templates
- [X] get Bob's .astylerc update
- [X] remember to snag issue / pull-request template changes from dummy repo
- [X] draft some Contributing guidelines
- [X] add blurb about semantic versioning in ReadMe
- [X] mark ready for review, without rebasing
- [x] try running src/run-manual-test on Windows
- [ ] rebase for clarity
- [ ] ?

owaddell-beckman committed:
release: swish 2.0.0

- bump Swish version to 2.0.0 to avoid confusion with internal 1.x
- set Swish version based on value in swish.h
- commonize refresh of compile-time includes in Mf-base
- use simpler form of include-line since Chez Scheme 9.5.4
  provides the revision information we need
- remove occurrences of "Swish Version" from documentation
- add change log
- add contributing guidelines
- add github issue / pull-request templates

owaddell-beckman committed:
swish-build: add --info for use in issue template

`swish-build --info` now reports information about the swish
executable's software environment, including operating-system
information and data from software-info

The test added here isn't ideal since it replicates code, but using
swish-build-test has the virtue of handling coverage.

owaddell-beckman committed:
add get-uname

owaddell-beckman committed:
add Chez Scheme machine-type to software-info

burgerrg committed:
astyle: make reference alignment the same as pointer alignment

burgerrg commented:
src/run-manual-test does this if all I've done is `./configure` and `make -C src/swish mat-prereq`:

```
DESKTOP-8N36EVU:~/swish$ src/run-manual-test 1
== Case 1
src/run-manual-test: line 25: data/tmp/manual-test-case: No such file or directory
manual-test-case: I/O error -4058 from uv_fs_realpath on data/tmp/manual-test-case: no such file or directory.
```

The script assumes the data/tmp directory has already been created.

owaddell-beckman committed:
test: fix manual test

The manual test assumed that the data/tmp directory already existed.

owaddell-beckman commented:
Good catch. Pushed an attempted fix.

burgerrg commented on doc/swish/app.tex:
```diff
@@ -82,21 +82,24 @@
-Here the \lopt{} option tells Swish to start a REPL after loading the script.
-\codebegin
+Here the \code{-q} option suppresses the startup message and sets the prompt to
+the empty string and the \lopt{} option tells Swish to start a REPL after
```
suggestion: add a comma between `string` and `and`.

burgerrg reviewed (commented):
That fixed src/run-manual-test, thanks!

owaddell-beckman committed:
doc: try to fix bad breaks

Add a "suggested" pagebreak at the start of each makedecl environment.
This seems to help avoid the otherwise horrible breaks we get around p.25
of swish.pdf.

owaddell-beckman committed:
log get-uname data in system_attributes on startup

Not marking this as a breaking change, though it adds several columns to the
system_attributes table. Marking 5f5ca59ea8a5d4efac3484b4df7d5154114767d5 as
a breaking change was probably misguided.

Developers should avoid "select *" queries in code, as that is fragile.

owaddell-beckman commented:
Sorry, I flipped it from a draft pull request prematurely. I think the changes I meant to include are now in place.

burgerrg reviewed (approved):
Great work, Oscar, thanks!

owaddell-beckman commented:
Pushed an attempt at rebase cleanup. I wouldn't mind squashing in the .gitignore fix to make it tidier when we tag the release.

owaddell-beckman committed:
test: update check-docs target

owaddell-beckman committed:
doc: add tupledef on <child> documentation

owaddell-beckman commented:
Pushed some minor documentation updates to include in 2.0.0. I plan to rebase these ahead of the "releae: swish 2.0.0" commit before we push the big red button.

owaddell-beckman commented:
Went ahead and squashed the .gitignore fix and reordered a few commits since I wanted to update some of the log messages.

owaddell-beckman commented:
Thanks, Bob and Chris, for your help in getting this release out the door.
